NickvisionTubeConverter.Shared/Models/MediaUrlInfo.cs

```diff
@@ -0,0 +1,122 @@
+        dynamic outFile = PythonHelpers.SetConsoleOutputFilePath(pathToOutput);
+        return await Task.Run(() =>
+        {
+            try
+            {
+                var mediaUrlInfo = new MediaUrlInfo(url);
+                using (Python.Runtime.Py.GIL())
+                {
+                    dynamic ytdlp = Python.Runtime.Py.Import("yt_dlp");
+                    var ytOpt = new Dictionary<string, dynamic>() {
+                        { "quiet", true },
+                        { "merge_output_format", "/" },
+                        { "windowsfilenames", RuntimeInformation.IsOSPlatform(OSPlatform.Windows) },
+                        { "ignoreerrors", true }
+                    };
+                    Python.Runtime.PyDict? mediaInfo = ytdlp.YoutubeDL(ytOpt).extract_info(url, download: false);
+                    if (mediaInfo == null)
+                    {
+                        outFile.close();
+                        return null;
+                    }
+                    if (mediaInfo.HasKey("entries"))
+                    {
+                        foreach (var e in mediaInfo["entries"].As<Python.Runtime.PyList>())
+                        {
+                            if (e.IsNone())
+                            {
+                                continue;
+                            }
+                            mediaUrlInfo.ParseFromPyDict(e.As<Python.Runtime.PyDict>(), true);
+                        }
+                    }
+                    else
+                    {
+                        mediaUrlInfo.ParseFromPyDict(mediaInfo);
+                    }
+                    outFile.close();
+                }
+                return mediaUrlInfo;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                using (Python.Runtime.Py.GIL())
+                {
+                    outFile.close();
+                }
+                return null;
+            }
+        });
+    }
+
+    private void ParseFromPyDict(Python.Runtime.PyDict mediaInfo, bool isPartOfPlaylist = false)
+    {
+        var title = mediaInfo.HasKey("title") ? (mediaInfo["title"].As<string?>() ?? "Media") : "Media";
+        foreach (var c in Path.GetInvalidFileNameChars())
+        {
+            title = title.Replace(c, '_');
+        }
+        foreach (var f in mediaInfo["formats"].As<Python.Runtime.PyList>())
+        {
+            var format = f.As<Python.Runtime.PyDict>();
+            if (format.HasKey("vbr"))
+            {
+                var resolution = new VideoResolution(format["width"].As<int>(), format["height"].As<int>());
+                if (!VideoResolutions.Exists(r => r == resolution))
+                {
+                    VideoResolutions.Add(resolution);
+                }
+            }
+        }
+        VideoResolutions.Sort();
+        VideoResolutions.Reverse();
```

Comment:
Since this is the only instance of a list of video resolutions being sorted, the video resolution CompareTo method should probably change so that the default sort order isn't in the reversed order of what we want

Reply:
> Since this is the only instance of a list of video resolutions being sorted, the video resolution CompareTo method should probably change so that the default sort order isn't in the reversed order of what we want

@nlogozzo Doesn't it interfere with the idea of models being independent of the rest of a program? Making `CompareTo` sorting from greater to smaller can be unexpected if used outside of context of TC.